pkg/virt-launcher/virtwrap/api/schema.go

```diff
@@ -745,22 +745,43 @@ type ConsoleSource struct {
 // BEGIN Inteface -----------------------------
 
 type Interface struct {
-	Address             *Address         `xml:"address,omitempty"`
-	Type                string           `xml:"type,attr"`
-	TrustGuestRxFilters string           `xml:"trustGuestRxFilters,attr,omitempty"`
-	Source              InterfaceSource  `xml:"source"`
-	Target              *InterfaceTarget `xml:"target,omitempty"`
-	Model               *Model           `xml:"model,omitempty"`
-	MAC                 *MAC             `xml:"mac,omitempty"`
-	MTU                 *MTU             `xml:"mtu,omitempty"`
-	BandWidth           *BandWidth       `xml:"bandwidth,omitempty"`
-	BootOrder           *BootOrder       `xml:"boot,omitempty"`
-	LinkState           *LinkState       `xml:"link,omitempty"`
-	FilterRef           *FilterRef       `xml:"filterref,omitempty"`
-	Alias               *Alias           `xml:"alias,omitempty"`
-	Driver              *InterfaceDriver `xml:"driver,omitempty"`
-	Rom                 *Rom             `xml:"rom,omitempty"`
-	ACPI                *ACPI            `xml:"acpi,omitempty"`
+	Address             *Address               `xml:"address,omitempty"`
+	Type                string                 `xml:"type,attr"`
+	TrustGuestRxFilters string                 `xml:"trustGuestRxFilters,attr,omitempty"`
+	Source              InterfaceSource        `xml:"source"`
+	Target              *InterfaceTarget       `xml:"target,omitempty"`
+	Model               *Model                 `xml:"model,omitempty"`
+	MAC                 *MAC                   `xml:"mac,omitempty"`
+	MTU                 *MTU                   `xml:"mtu,omitempty"`
+	BandWidth           *BandWidth             `xml:"bandwidth,omitempty"`
+	BootOrder           *BootOrder             `xml:"boot,omitempty"`
+	LinkState           *LinkState             `xml:"link,omitempty"`
+	FilterRef           *FilterRef             `xml:"filterref,omitempty"`
+	Alias               *Alias                 `xml:"alias,omitempty"`
+	Driver              *InterfaceDriver       `xml:"driver,omitempty"`
+	Rom                 *Rom                   `xml:"rom,omitempty"`
+	ACPI                *ACPI                  `xml:"acpi,omitempty"`
+	Backend             *InterfaceBackend      `xml:"backend,omitempty"`
+	PortForward         []InterfacePortForward `xml:"portForward"`
```

Comment:
I followed the example from libvirt-go-xml-module implementation
https://gitlab.com/libvirt/libvirt-go-xml-module/-/blob/master/domain.go?ref_type=heads#L839

For what I undersnad because its a slice it will interpreted similar to pointers, if PortForward is not specified it will be nil. 
Do you think it should be set with `omitempty` anyways?

Reply:
It seems there is no difference if `omitempty` is specified for pointers / "reference objects" (e.g.: slices).
Such attributes will be marshaled similarly , if they are not specified no tag is generated.
See this example https://go.dev/play/p/AyLDAaacL78